File: Backend/API-Data/src/Model/ParcelaModel.cs
using API_Data.src.Enum;
using System.ComponentModel.DataAnnotations.Schema;

namespace API_Data.src.Model
{
    public class Parcela
    {
    
        public int Id { get; set; }

   
        public int NumeroParcela { get; set; }

  
        public decimal ValorParcela { get; set; }

     
        public DateTime DataVencimento { get; set; }


        public DateTime? DataPagamento { get; set; }


        public StatusParcela Status { get; set; } = StatusParcela.Aberto;

        public int LancamentoId { get; set; }
        public Lancamento Lancamento { get; set; } = null!;
    }
}

File: Frontend/My-finances-Web/src/app/models/parcela-model.ts
import { StatusParcelaEnum } from '../enums/status-parcela-enum';

export interface returnParcela {
  id: number;
  dependence_id: number;
  descricao: string;
  valorParcela: number;
  dataVencimento: string;
  status: StatusParcelaEnum;
  atribuicao: number;
  numeroParcela: number;
}